examples/catalyst_contrastive/barlow_twins.py

```diff
@@ -0,0 +1,169 @@
+# flake8: noqa
+import argparse
+from functools import partial
+
+from sklearn.linear_model import LogisticRegression
+from sklearn.metrics import top_k_accuracy_score
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.optim as optim
+from torch.utils.data import DataLoader
+import torchvision
+from torchvision import transforms
+from torchvision.models.resnet import resnet50
+
+from catalyst import dl
+from catalyst.contrib.models.cv.encoders import ResnetEncoder
+from catalyst.contrib.nn import BarlowTwinsLoss
+
+
+class CifarPairTransform:
+    def __init__(self, train_transform=True, pair_transform=True):
+        if train_transform is True:
+            self.transform = transforms.Compose(
+                [
+                    transforms.RandomResizedCrop(32),
+                    transforms.RandomHorizontalFlip(p=0.5),
+                    transforms.RandomApply([transforms.ColorJitter(0.4, 0.4, 0.4, 0.1)], p=0.8),
+                    transforms.RandomGrayscale(p=0.2),
+                    transforms.ToTensor(),
+                    transforms.Normalize([0.4914, 0.4822, 0.4465], [0.2023, 0.1994, 0.2010]),
+                ]
+            )
+        else:
+            self.transform = transforms.Compose(
+                [
+                    transforms.ToTensor(),
+                    transforms.Normalize([0.4914, 0.4822, 0.4465], [0.2023, 0.1994, 0.2010]),
+                ]
+            )
+        self.pair_transform = pair_transform
+
+    def __call__(self, x):
+        if self.pair_transform is True:
+            y1 = self.transform(x)
+            y2 = self.transform(x)
+            return y1, y2
+        else:
+            return self.transform(x)
+
+
+class Model(nn.Module):
+    def __init__(self, feature_dim=128, **resnet_kwargs):
+        super(Model, self).__init__()
+        # encoder
+        self.f = ResnetEncoder(**resnet_kwargs)
+        # projection head
+        self.g = nn.Sequential(
+            nn.Linear(2048, 512, bias=False),
+            nn.BatchNorm1d(512),
+            nn.ReLU(inplace=True),
+            nn.Linear(512, feature_dim, bias=True),
+        )
+
+    def forward(self, x):
+        x = self.f(x)
+        feature = torch.flatten(x, start_dim=1)
```

Comment:
could we use `nn.Flatten` during `self.x = nn.Sequential(ResnetEncoder(), nn.Flatten())`?
for the best PyTorch practices, it's preferable to use `nn.Modules` instead of Functional API calls

we also have an "old-pytorch" implementation https://github.com/catalyst-team/catalyst/blob/master/catalyst/contrib/nn/modules/common.py#L8

Reply:
Good point. 
[Done](https://github.com/catalyst-team/catalyst/pull/1261/commits/6f3e25f008c5cada5de0b22f5429944caab68519)